src/System.Security.Cryptography.Xml/tests/EncryptionMethodTests.cs

```diff
@@ -0,0 +1,167 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Xml;
+using Xunit;
+
+namespace System.Security.Cryptography.Xml.Tests
+{
+    public class EncryptionMethodTests
+    {
+        [Fact]
+        public void Ctor_Default()
+        {
+            EncryptionMethod method = new EncryptionMethod();
+            Assert.Equal(0, method.KeySize);
+            Assert.Null(method.KeyAlgorithm);
+        }
+
+        [Theory]
+        [InlineData(null)]
+        [InlineData("")]
+        [InlineData("algorithm")]
+        public void Ctor_String(string algorithm)
+        {
+            EncryptionMethod method = new EncryptionMethod(algorithm);
+            Assert.Equal(0, method.KeySize);
+            Assert.Equal(algorithm, method.KeyAlgorithm);
+        }
+
+        [Theory]
+        [InlineData(1)]
+        [InlineData(int.MaxValue)]
+        public void KeySize_Set_SetsValue(int value)
+        {
+            EncryptionMethod method = new EncryptionMethod() { KeySize = value };
+            Assert.Equal(value, method.KeySize);
+        }
+
+        [Theory]
+        [InlineData(-1)]
+        [InlineData(0)]
+        public void KeySize_SetNegativeValue_ThrowsArgumentOutOfRangeException(int value)
+        {
+            EncryptionMethod method = new EncryptionMethod();
+            Assert.Throws<ArgumentOutOfRangeException>("The key size should be a non negative integer.", () => method.KeySize = value);
```

Comment:
This is a really long line, and the message verification is only correct when using the en-us resources.

Though, I think that's supposed to be the parameter name, meaning we have a buggy exception :smile:.